crates/wysiwyg/src/composer_model/join_nodes.rs

```diff
@@ -271,7 +299,8 @@ where
         &mut self,
         from_handle: &DomHandle,
         to_handle: &DomHandle,
-    ) -> usize {
+    ) -> (usize, HashMap<DomHandle, DomHandle>) {
```

Comment:
I don't really like the idea of returning the moved handles here, but I couldn't find a better way to do it. Suggestions are welcome.

Reply:
We could pass in a mutable reference similar to how we are in join_format_nodes_at_level?

Reply:
To me it makes more sense to just return the value and it's up to the user to actually use it or not rather than forcing us to pass an empty HashMap just in case we want to use them later. I'd use an `Option<&mut HashMap>` here, but we've already seen that passing mutable references inside Options is no easy feat 😅 .